### ISSUE TITLE
use partykit functions for mmif?

### ISSUE THREAD

**Issue by tdhock** [aldro61/mmit#40]
hey @parismita I noticed that you coded the mmif function "from scratch" and that is fine. But I was wondering if there was some simpler way to do it using functions provided by partykit. I asked Torsten:
```
tdhock> does partykit provide some helper functions to help automate / simplify fitting an
tdhock> ensemble?

there is some inftastructure (used by partykit::cforest or
trtf::traforest) but it is not as rich as the tree structure. At least you
can get node predictions and prediction weight without having to worry.
And there is some code for variable importances with user-defined loss
functions.
```
so it may be useful to look into how those functions are implemented if we want to change/simplify our implementation of random forests.

No need to do this right away but I thought I would mention that here, in case we ever want to.

**Comment by parismita** [comment]
yeah, I'll look into it...it would be cool to use some partykit or trtf function rather than coding from scratch.
